docs/Documentation/SystemDesign/5-DataQuery/6-LastQuery.md

```diff
@@ -0,0 +1,122 @@
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+    
+        http://www.apache.org/licenses/LICENSE-2.0
+    
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# Last query
+
+The main logic of Last query is in LastQueryExecutor
+
+* org.apache.iotdb.db.query.executor.LastQueryExecutor
+
+The Last query executes the `calculateLastPairForOneSeries` method for each specified time series.
+
+## Read MNode cache data
+
+We add a Last data cache to the MNode structure corresponding to the time series that needs to be queried.
+
+`calculateLastPairForOneSeries` method For the last query of a certain time series, first try to read the cached data in the MNode.
+
+```
+try {
+  node = MManager.getInstance().getDeviceNodeWithAutoCreateStorageGroup(seriesPath.toString());
+} catch (MetadataException e) {
+  throw new QueryProcessException(e);
+}
+if (((LeafMNode) node).getCachedLast() != null) {
+  return ((LeafMNode) node).getCachedLast();
+}
+```
+If it is found that the cache has not been written, execute the following standard query process to read the TsFile data.
+
+## Last standard query process
+
+Last standard query process needs to traverse all sequential files and out-of-order files to get query results, and finally write the query results back to the MNode cache.  In the algorithm, sequential files and out-of-order files are processed separately.
```

Comment:
It is better to replace "out-of-order" with "unsequential files" or just "unseq files". It is like a idiom to use "unsequential files" to represent the unordered files in our related documents.